### ISSUE TITLE
YoloV5 Detect Layer.

### ISSUE THREAD

**Issue by Dharmil95** [marcoslucianops/DeepStream-Yolo#78]
Have you converted the Detect() class as it is from the ultralytics repo or you have made some changes in it? Following is the snippet of the code used in ultralytics for Final Detect layer.

`def forward(self, x):
        for i in range(len(x)):
            print(x[i].shape)
        print("GRID: ", self.grid)
        print((x, self.no * self.na, 1) for x in ch)
        exit(-1)
        # x = x.copy()  # for profiling
        z = []  # inference output
        for i in range(self.nl):
            x[i] = self.m[i](x[i])  # conv
            bs, _, ny, nx = x[i].shape  # x(bs,255,20,20) to x(bs,3,20,20,85)
            x[i] = x[i].view(bs, self.na, self.no, ny, nx).permute(0, 1, 3, 4, 2).contiguous()

            if not self.training:  # inference
                if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:
                    self.grid[i] = self._make_grid(nx, ny).to(x[i].device)

                y = x[i].sigmoid()
                if self.inplace:
                    y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + self.grid[i]) * self.stride[i]  # xy
                    y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i]  # wh
                else:  # for YOLOv5 on AWS Inferentia https://github.com/ultralytics/yolov5/pull/2953
                    xy = (y[..., 0:2] * 2. - 0.5 + self.grid[i]) * self.stride[i]  # xy
                    wh = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i].view(1, self.na, 1, 1, 2)  # wh
                    y = torch.cat((xy, wh, y[..., 4:]), -1)
                z.append(y.view(bs, -1, self.no))

        return x if self.training else (torch.cat(z, 1), x)`

**Comment by marcoslucianops** [comment]
I used the TensorRTX repo files to make the YOLOv5 works in DeepStream. I'm not sure about the functions.
